cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -175,17 +175,18 @@ private ProgramStatus getExistingAppProgramStatus(ApplicationSpecification appSp
         }
         ensureAccess(programId);
 
-        if ((programId.getType() == ProgramType.MAPREDUCE || programId.getType() == ProgramType.SPARK) &&
-          !store.getRuns(programId, ProgramRunStatus.RUNNING, 0, Long.MAX_VALUE, 1).isEmpty()) {
-          // MapReduce program exists and running as a part of Workflow
-          return ProgramStatus.RUNNING;
-        }
-        return ProgramStatus.STOPPED;
+        // If no runtime info exists for a program, the program status can be determined based on if the store has
+        // a RUNNING run record or not
+        return (store.getRuns(programId, ProgramRunStatus.RUNNING, 0, Long.MAX_VALUE, 1).isEmpty()) ?
```

Comment:
the mapping here is not the same as the mapping in `ProgramRunSTatus.toProgramStatus`, right?